### DISCUSSION TITLE
PA2: about sendPacket() function

### DISCUSSION THREAD

**Discussion by overflow218** [ANLAB-KAIST/KENSv3#60] [Project Q&A 2. Transport Layer (TCP)] [answered]
Dear TA, 
our team is stuck on sendPacket()
In connect() or accept() , we are trying to send packet with sendPacket(). 
we also make a packet  and consider follow things
- room for Ethernet, Ip headers
- Source and Destination Ip address
- TCP header packet format like source port, dest port, sequence number, ack number, SYN, ACK etc ...
we double checked packet data after writing data to packet and re-reading from packet. 
It seems go somewhere but anything happens (T-T)
We even tried to send back packet arrived at PacketArrived by changing  source IP addr and dest IP addr, and change 
sequence number, ack number. but It doesn't work...
We spent so much time on it.  what should we do?
Is there any way that we can check our packet arrived well except timer or something?

-----------
안녕하세요 조교님 
sendPacket()함수를 이용해 패킷을 보내는것을 시도중입니다. 
답변해주신 것과 같이  Ethernet, Ip headers 부분도 고려해서 패킷을 만들었고 
교과서를 참고하여 TCP header packet에 써야한다고 생각하는 부분은 다 작성해서 패킷을 만들었습니다. 
패킷에 쓴 데이터를 다시 읽어와서 데이터가 잘 들어갔는지도 확인했습니다. 
그런데 sendPacket()을 이용해서 보내고나면 아무 일도 일어나지 않습니다 ㅠㅠ ( 다시 응답을 해줘야하는 connect()는 제외하더라도 
listen(), accept()를 확인하는 부분에서 (반대편에서) 패킷을 잘 받지 못했는지 테스트가 실패합니다.)
뭔가 실수가 있었는지 확인하기 위해 PacketArrived()에 들어온 패킷을 소스 아이피 주소와 목적지 아이피 주소를 바꾸고 
sequence number, ack number만 수정해서 보내봤는데도 별다른 응답이 없습니다... 
어떻게 하면 이 부분을 해결할 수 있을까요? 타이머 말고도 켄즈상에서 보낸 패킷이 잘 도착했는지 확인할 수 있는 방법이 있는지 궁금합니다.
감사합니다.

**Comment by GrasshopperBears** [comment, marked as answer]
테스트가 실행되면 생성되는 ${TEST_NAME}.pcap 파일을 wireshark로 열면 주고받은 패킷을 확인하실 수 있습니다
